unblob/processing.py

```diff
@@ -55,6 +61,13 @@ def process_file(config: ExtractionConfig, path: Path) -> List[Report]:
         depth=0,
     )
 
+    if config.extract_root.exists() and list(config.extract_root.iterdir()):
+        if config.force_overwrite_extract_dir:
+            logger.debug("Overwriting extract root directory")
+            rmtree(config.extract_root)
+        else:
+            raise ExtractDirNotEmptyException
+
```

Comment:
I think it is enough to check if the directory exists or not, I do not think we need to check for if it is empty: it will be [re-]created anyway.

If the above is agreed upon, the exception `ExtractDirNotEmptyException` should also be ranamed to `ExtractDirAlreadyExistsError` (`Error` is the usual suffix for Python exceptions).

Reply:
I don't know how the end user is expecting to use this tool. In my mind, it is a perfectly acceptable usecase to have an empty directory created and use that for extraction directory. In that case, it would be annoying to have an error to delete that empty dir, so the tool could re-create it...

But on the other hand, your solution sounds simpler.

@QKaiser / @vlaci, what do you think, how should it be? (I know this is not a big issue, but I would rather decide it here in the scope of this pull request than modifying this later on)

Reply:
I don't think, we can expect the user to figure out our directory naming scheme and create output directories ahead of time.

Reply:
I'm with @vlaci on this one.

Reply:
Of course, but this option is only for the extract root dir (`-e, --extract-dir DIRECTORY` parameter), not for every subsequent weird-named sub-directory :sweat_smile:

Reply:
I am confused, as extract root dir doesn't need to be empty